textproto/header_test.go

```diff
@@ -404,18 +404,24 @@ var formatHeaderFieldTests = []struct {
 	{
 		k:         "Subject",
 		v:         "=?utf-8?q?=E2=80=9CDeveloper_reads_customer_requested_change.=E2=80=9D=0A?= =?utf-8?q?=0ACaravaggio=0A=0AOil_on...?=",
-		formatted: "Subject: =?utf-8?q?=E2=80=9CDeveloper_reads_customer_requested_change.\r\n =E2=80=9D=0A?= =?utf-8?q?=0ACaravaggio=0A=0AOil_on...?=\r\n",
+		formatted: "Subject: =?utf-8?q?=E2=80=9CDeveloper_reads_customer_requested_change.=E2=80=9D=0A?= =?utf-8?q?=0ACaravaggio=0A=0AOil_on...?=\r\n",
 	},
+	// Spaces should not appear in RFC2047-encoded string but it should be fine in case they do
 	{
 		k:         "Subject",
 		v:         "=?utf-8?q?=E2=80=9CShort subject=E2=80=9D=0A?= =?utf-8?q?=0AAuthor=0A=0AOil_on...?=",
-		formatted: "Subject: =?utf-8?q?=E2=80=9CShort subject=E2=80=9D=0A?= =?utf-8?q?\r\n =0AAuthor=0A=0AOil_on...?=\r\n",
+		formatted: "Subject: =?utf-8?q?=E2=80=9CShort subject=E2=80=9D=0A?=\r\n =?utf-8?q?=0AAuthor=0A=0AOil_on...?=\r\n",
```

Comment:
Hmm. What if a space appears in a RFC2047-encoded string and we fold it? e.g. `=?utf-8?q?=E2=80=9CShort subject=E2=80=9D=0A?=` → `=?utf-8?q?=E2=80=9CShort\r\n subject=E2=80=9D=0A?=`

Are we allowed to do this? IIRC we had the quoted-printable regex to avoid this.

Reply:
1. It should not happen because it's clearly an ill-formed word (2 and 6.3 of RFC 2047).
2. When unfolded it should return back to it's original form, it should be decoded the same way (if possible). To my knowledge Golang mime.WordDecoder and Thunderbird decoder both can handle spaces in Q-encoded utf-8 strings.

Reply:
Hmm. Seems like it indeed, should've investigated a little bit more before merging https://github.com/emersion/go-message/pull/22 (which this PR effectively reverts).

Anyway, I agree with you, this PR does the right thing.

Reply:
Going through RFC and possible cases I've found something possibly dangerous in `\n` handling. 

If input contains `\n` without corresponding `\r` we will have an incorrect output (neither printable ASCII, HTAB or SPACE, not `\r\n` line terminator) . I'm not sure about practical possibilities of this though since any encoder should take care of this `\n` *beforehand*

Reply:
Users aren't supposed to provide header keys/values with `\n` or `\r`. Maybe we should error out in this case?

Reply:
(This is a separate change though - should probably be in a separate commit or PR)

Reply:
Sure error is the right thing here.
I may open a new issue to discuss possible implementations, or if you have an idea right now, or if you're able to fix it straight away, you'll likely do it better since it's your code.

Reply:
Yes, please open an issue! I always prefer to help contributors send patches instead of doing it myself, that's healthier for the project on the long run.